### PR TITLE
Add Rule for SLES-15-010140  & SLES-12-010100

### PR DESCRIPTION
**Pull request by yarunachalam**
#### Description:

- Add Rule for SLES-15-010140  & SLES-12-010100

#### Rationale:

- Add SLE12 & SLE15 stig to 'Implement Blank Screensaver'

### CODE REVIEW

**Comment by vojtapolasek** [comment]
I see that you made existing remediations applicable for SLE. In rule.yml, you are effectively saying that in case of SLE, this is configured at user level. But remediations configure it system-vide. Is this what you want?

**Comment by brett060102** [comment]
@vojtapolasek yes, this is what we want. The changes at the system level will create the correct default environment, but it is still possible for users to override this. So, our the STIGs call out both the system settings and user action. We wanted that to be represented in the description.

**Comment by vojtapolasek** [comment]
@brett060102  I understand, but in the current state, the description and remediation differ, I think this can be confusing. How about keeping this sle-specific statement, but appending the original one as well? You will have in the dwescription the steps for user.specific config as well as default config.

**Comment by brett060102** [comment]
@vojtapolasek sounds good. I will work with @yarunachalam to get that done.

**Comment by yarunachalam** [comment]
> @vojtapolasek sounds good. I will work with @yarunachalam to get that done.

@vojtapolasek re-worked the description text, Does this look good to you?

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
 title: 'Implement Blank Screensaver'
 
 description: |-
+    {{# The SLE12 and SLE15 STIG unfortunately requires per-user settings, #}}
+    {{# not a globally locked one, so we don't require but only suggest the latter. #}}
+    {{% if product in ["sle12","sle15"]%}}
+    On SUSE users should set the screensaver to use publicly viewable images or blank screen
+    by doing the following to configure the system.  Find the
+    Settings menu and then navigate to the Background selection section:
+    <tt>- Click "Activities" on the top left.</tt>
```
Comment (brett060102) [inline linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:14]: @yarunachalam I am sorry I should have caught this The directions are slightly different, so I think that this block needs to be:
   {{% if product == "sle15" }}
   <tt>- Click "Activities" on the top left.</tt>
   <tt>- Click "Show Applications" at the bottom of the Activities menu.</tt>
   <tt>- Click the "Settings" icon.</tt>
   <tt>- Click "Background" from left hand menu.</tt>
   <tt>- Select image and set the Lock Screen image to the user's choice.</tt>
   <tt>- Exit Settings Dialog.</tt>
   {{%- else %}}
   <tt>- - Click "Applications" on the bottom left.</tt>
   <tt>- - Hover over "System Tools" with the mouse.</tt>
   <tt>- - Click the "Settings" icon under System Tools.</tt>
   <tt>- - Click "Background" and then "Lock Screen".</tt>
   <tt>- Select image and set the Lock Screen image to the user's choice.</tt>
   <tt>- Click "Select".</tt>
   <tt>- Exit Settings Dialog.</tt>
   {{%- endif %}}

Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:11]: Fixed the description text.

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
 
-prodtype: fedora,ol7,ol8,rhel7,rhel8
+prodtype: fedora,ol7,ol8,rhel7,rhel8,sle12,sle15
 
 title: 'Implement Blank Screensaver'
 
 description: |-
+    {{# The SLE12 and SLE15 STIG unfortunately requires per-user settings, #}}
+    {{# not a globally locked one, so we don't require but only suggest the latter. #}}
+    {{% if product == "sle15" }}
```
Comment (brett060102) [inline linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:10]: The block:
    {{% if product in ["sle12","sle15"]%}}
    On SUSE users should set the screensaver to use publicly viewable images or blank screen
    by doing the following to configure the system.  Find the
    Settings menu and then navigate to the Background selection section:

should have remained,
So we have stuff that applies to both sl15 and sle12, with in that there info that is specific to sl15 and other info that is sle12 specific.

Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:14]: Fixed

**Comment by brett060102** [comment]
@vojtapolasek It was a winding road, but this should be good to go now.

**Comment by brett060102** [comment]
@vojtapolasek Is there an issue with this PR that we need to address?

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml [hunk 3]
```diff
@@ -26,13 +49,21 @@ severity: medium
...
+    cce@sle12: CCE-83221-2
+    cce@sle15: CCE-85715-1
 
 references:
     cjis: 5.5.5
     cui: 3.1.10
     disa: CCI-000060
     nist: AC-11(1),CM-6(a)
     nist-csf: PR.AC-7
+    nist@sle15: AC-11(1),AC-11(1).1
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:61]: Please condense these two nist@ into one e.g.
```suggestion
    nist: AC-11(1),AC-11(1).1
```

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml [hunk 4]
```diff
@@ -26,13 +49,21 @@ severity: medium
...
+    nist@sle12: AC-11(1),AC-11(1).1
+    stigid@sle15: SLES-15-010140
+    stigid@sle12: SLES-12-010100
+    srg@sle15: SRG-OS-000031-GPOS-00012
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml:65]: Please merge these srgs into one, srgs are product independent, e.g.:
```suggestion
    srg: SRG-OS-000031-GPOS-00012
```

**Comment by vojtapolasek** [comment]
Please just fix these references and we are good to go.
Note that disa, srg and nist are product independent, no need for @sle12 there.

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml (hunk 4 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml]: done

File: linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml (hunk 3 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_screen_locking/dconf_gnome_screensaver_mode_blank/rule.yml]: Done

**Comment by yarunachalam** [comment]
> Please just fix these references and we are good to go.
> Note that disa, srg and nist are product independent, no need for @sle12 there.

Fixed.

**Comment by vojtapolasek** [comment]
Merging, the rule is not OCP related.

**Pull request merged**